core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2979,7 +2979,7 @@ public void setAttribute(AlluxioURI path, SetAttributeContext context)
     boolean rootRequired = options.hasOwner();
     // for chgrp, chmod
     boolean ownerRequired =
-        (options.hasGroup()) || (options.hasMode() && options.getMode() != Constants.INVALID_MODE);
+        (options.hasGroup()) || (options.hasMode());
```

Comment:
Can be on one line? Also if we have a mode is it impossible for it to be invalid?

Reply:
Correct. I made the bit fields in PMode mandatory.